### PR TITLE
EVA-2913 - First version of the contig name converter with support for VCF files

### PR DESCRIPTION
**Pull request by tcezard**

### CODE REVIEW

File: tests/test_convert_vcf_file.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+               os.remove(f)
+
+    def last_line(self, file_to_check):
+        with open(file_to_check) as open_file:
+            for line in open_file:
+                pass
+        return line
+
+    def test_convert_to_ena(self):
+        input_file = os.path.join(os.path.dirname(__file__), 'resources/test.vcf')
```
Comment (apriltuesday) [inline tests/test_convert_vcf_file.py:23]: Should commit the test file as well.

**Review by apriltuesday — COMMENTED** [review]
For consistency with the rest of our repos, could we change the name to `contig-name-converter`?

(Also I think you made some updates right before I submitted this review, hopefully most of my comments are still applicable...)

File: tests/test_convert_vcf_file.py (hunk 1 above)
Comment (tcezard) [reply tests/test_convert_vcf_file.py:23]: Done

File: contig_name_converter/convert_vcf_file.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    converter = SequenceAccessionConverter(target_naming_convention)
+
+    contig_name_to_accession = {}
+    # Get a new empty header
+    output_header = VariantHeader()
+    # Fill it with all the values from the input
+    for header_rec in input_file.header.records:
+        if header_rec.type != 'CONTIG':
+            output_header.add_record(header_rec)
+            pass
```
Comment (sundarvenkata-EBI) [inline contig_name_converter/convert_vcf_file.py:26]: Is this needed here?

File: contig_name_converter/converters.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+class SequenceAccessionConverter:
+    """Converter that needs to receive the INSDC contig accession that can uniquely identify the sequence to
+    translate the contig name"""
+    contig_alias_url = 'https://wwwdev.ebi.ac.uk/eva/webservices/contig-alias/'
+
+    def __init__(self, target_naming_convention):
+        self.target_naming_convention = target_naming_convention
+        self._cache = {}
+
+    @retry(tries=3)
```
Comment (sundarvenkata-EBI) [inline contig_name_converter/converters.py:14]: suggest adding jitter and backoff parameters as well...

**Review by sundarvenkata-EBI — APPROVED** [review]

File: .github/workflows/python-package.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.8]
+        java-version: [1.8]
```
[the remark this answers is not in the document]
Comment (tcezard) [reply .github/workflows/python-package.yml]: Removed

File: convert_vcf_file.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        rec.translate(output_header)
+        output_file.write(rec)
+
+
+def main():
+    argparse = ArgumentParser(description="Convert sequence/contig names from one convention to another.")
+    argparse.add_argument('-i', '--input', help='Input file to convert')
+    argparse.add_argument('-o', '--output', help='Output file containing convert the converted data')
+    argparse.add_argument('-c', '--convention', help='Contig naming convention use the.',
+                          choices=['enaSequenceName', 'genbank', 'genbankSequenceName', 'refseq', 'ucscName'],
```
[the remark this answers is not in the document]
Comment (tcezard) [reply convert_vcf_file.py]: Done

File: converters.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
+import requests
+from retry import retry
+
+
+class SequenceAccessionConverter:
+    """Converter that needs to receive the INSDC contig accession that can uniquely identify the sequence to
+    translate the contig name"""
+    contig_alias_url = 'https://wwwdev.ebi.ac.uk/eva/webservices/contig-alias/'
```
[the remark this answers is not in the document]
Comment (tcezard) [reply converters.py]: Done

File: contig_name_converter/converters.py (hunk 3 above)
Comment (tcezard) [reply contig_name_converter/converters.py]: Done

File: contig_name_converter/convert_vcf_file.py (hunk 2 above)
Comment (tcezard) [reply contig_name_converter/convert_vcf_file.py]: removed

**Pull request merged**